lec16-data-viz.Rmd

```diff
@@ -1,4 +1,97 @@
 ---
-title: "More on data visualizations (plus project work)"
+title: "More on data visualizations and the git workflow (plus project work)"
+author: Joel Ostblom
 ---
 
+First, let's create a chat room for your repository. We will use gitter, since
+it has good integrations with GitHub. Got to http://gitter.im/, login with your
+GitHub account and click the plus sign to create a new chat room. Browse to
+your repository, and add the members of your team. Since your repository is
+public, this chatroom will also be public (so you might not want to share
+personal information, and keep your discussions on topic, just as if this was
+a real-world work project).
+
+Discussions in the chat room are good to crystallize what tasks need to be
+tackled next. Once you have identified a specific task, open an issue about it
+on GitHub to track the progress of that particular tasks. You should create
+issues for your planned analyses, writing, literature review, etc. Then
+schedule those into your day and close them when you are done with that task.
+It is similar to a powerful ToDo manager for your project. In general, you
+should create issues in the main repository, not in your own fork of the
+repository. Let's create a new issue based on our discussion, assign the person
+we think should work on it and then review the workflow for addressing the issue.
+
+*Create new issue about adding a file to the sandbox repo*
+
+1. Create your own fork of the online git repository via the GitHub interface.
+2. Click the green `Clone or download` button to copy the git URL.
+3. In RStudio, open a new git project.
+    - `File -> Version Control -> Git`. Paste the URL you copied previously and select where to save the directory.
+4. Click `Create Project`
+    - You can now see the new directory if you navigate to it in your file
+      navigator.
+5. Open a terminal and navigate to the new directory.
+    - You could do the navigation in your file browser and then right click to
+      open a terminal
+        - Right click should be enabled by default with Git.. on Windows. For
+          Mac's Finder, follow [these
+          instructions](https://stackoverflow.com/a/7054045/2166823) (in brief
+          `System Preferences > Keyboard > Shortcuts > Services`).
+6. In RStudio, click `History` under the small Git button menu.
+    - This is similar to typing `git log --oneline --graph` in terminal.
```

Comment:
I have left out how to make sure the fork is in sync with upstream. This is important, but potentially also a bit confusing. Is there an easier way (maybe through R studio) than to have them add the upstream remote, pull upstream to local master and potentially end up in vim to resolve a merge conflict?

Reply:
I believe you can pull from upstream using RStudio, but I don't recall if that is available only in the latest version. I normally just use the command line for that.